bsp/src/mill/bsp/BSP.scala

```diff
@@ -24,9 +24,11 @@ object BSP extends ExternalModule with CoursierModule with BspServerStarter {
   }
 
   private def bspWorkerLibs: T[Agg[PathRef]] = T {
-    resolveDeps(T.task {
-      bspWorkerIvyDeps().map(bindDependency())
-    })()
+    mill.modules.Util.millProjectModule(
+      "MILL_BSP_WORKER",
+      "mill-bsp-worker",
+      repositoriesTask()
+    )
```

Comment:
I can see how it improves the testability, but I'd rather get rid of the `millProjectModule` _hack_ altogether. It weakens the way Mill consumes project modules / plugins, and may be even a security vulnerability. Just using a local repo in the process of testings _is definitely_ more time consuming, but it disables this attack vector and also runs tests under more realistic conditions.

Reply:
I don't think I'm willing to give up purely local testing. The test pyramid is a pyramid, with both faster/lower-fidelity tests at the bottom and slower/higher-fidelity tests at the top. In this case we basically get both cases for free.

If there's a security concern, we should spell it out and find some way to mitigate it, but I'm sure that we can find a solution without giving up the ability to do local testing entirely

Reply:
Yeah, that's understandable. We could hook into the resolve instead, and just feed all  known jars directly. The potential injection is the smaller concern for me, it's rather the fact that we load the jars in a complete different way; e.g. when resolving via coursier, there might come different transitive deps as those we assemble in our build. 

But this is a orthogonal issue, so let's not continue this thinking here. I'll open a placeholder dicussion, where we can outline alternatives which are easier to maintain as the current setup in `build.sc`, which I don't like so much.